entity-service/src/main/kotlin/com/egm/stellio/entity/web/EntityAccessControlHandler.kt

```diff
@@ -394,4 +400,82 @@ class EntityAccessControlHandler(
 
         return ResponseEntity.status(HttpStatus.NO_CONTENT).build<String>()
     }
+
+    @GetMapping(produces = [MediaType.APPLICATION_JSON_VALUE, JSON_LD_CONTENT_TYPE])
+    suspend fun getEntitiesIdsHaveRights(
+        @RequestHeader httpHeaders: HttpHeaders,
+        @RequestParam params: MultiValueMap<String, String>
+    ): ResponseEntity<*> {
+        val count = params.getFirst(QUERY_PARAM_COUNT)?.toBoolean() ?: false
+        val (offset, limit) = extractAndValidatePaginationParameters(
+            params,
+            applicationProperties.pagination.limitDefault,
+            applicationProperties.pagination.limitMax,
+            count
+        )
+        val type = params.getFirst(QUERY_PARAM_TYPE)
+        var q = params.getFirst(QUERY_PARAM_FILTER)
+        val contextLink = getContextFromLinkHeaderOrDefault(httpHeaders)
+        val mediaType = getApplicableMediaType(httpHeaders)
+        val includeSysAttrs = params.getOrDefault(QUERY_PARAM_OPTIONS, emptyList())
+            .contains(QUERY_PARAM_OPTIONS_SYSATTRS_VALUE)
+        val useSimplifiedRepresentation = params.getOrDefault(QUERY_PARAM_OPTIONS, emptyList())
+            .contains(QUERY_PARAM_OPTIONS_KEYVALUES_VALUE)
+        val sub = getSubFromSecurityContext()
+        val expandedAttrs = parseAndExpandRequestParameter(null, contextLink)
+
+         if (q!=null &&
+                 (!q.equals(AUTH_TERM_CAN_READ) ||
+                !q.equals(AUTH_TERM_CAN_WRITE) ||
+                !q.equals(AUTH_TERM_CAN_ADMIN)))
```

Comment:
I remove this complex condition to `if (q!=null && !listOf(AUTH_TERM_CAN_READ,AUTH_TERM_CAN_WRITE,AUTH_TERM_CAN_ADMIN).contains(q))`